#1573: Default encoding of R Markdown file created by RTVS cause gar…

MikhailArkhipov opened the pull request:
…bling of non-ascii characters

int19h commented on src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs:
```diff
@@ -28,7 +29,7 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             string outputFile = Path.GetFileName(outputFilePath);
             string outputFolder = Path.GetDirectoryName(outputFilePath).Replace('\\', '/');
             // Run rmarkdown::render
-            return $"\"rmarkdown::render(\'{inputFile}\', output_format=\'{format}\', output_file='{outputFile}',  output_dir='{outputFolder}')\"";
```
What if they try to publish an existing, non-UTF .Rmd?

MikhailArkhipov replied:
This is generated command that we invoke so we specify encoding that we save the file with. However, it is a good question if someone tries to do the same manually. In this case they won't be going via this path and rather would need to a) save file in VS with desired encoding and b) specify that encoding in the command line. Although I guess no one would ever publish RMD manually.

MikhailArkhipov replied:
Alternatively, we can try and see if file is convertible to Encoding.Default and leave it as it as if it is convertable.

int19h replied:
If we control the encoding here, and this is not intended for direct user consumption, I wouldn't bother.

MikhailArkhipov replied:
Well, there is a side effect since if file is modified, we do save it forcibly in UTF-8 which may not be what the user wants. Let me try and add the detection.

int19h replied:
But if we save it using native encoding (when it can be thus saved), won't the original reported issue re-manifest?

Also, note that OS locale and R locale do not necessarily match (which was a root cause of many of our bugs regarding Unicode).

int19h commented:
:+1:

MikhailArkhipov commented:
Added detection if file can be converted. It appears that rmarkdown always produces UTF-8 page. However, original encoding has to be specified. Added the warning that file is not convertible and if user is OK with re-saving file in UTF-8

int19h commented:
:+1:
